src/transformers/modeling_utils.py

```diff
@@ -4906,6 +4940,12 @@ def _load_pretrained_model(
         if hf_quantizer is not None:
             expected_keys = hf_quantizer.update_expected_keys(model_to_load, expected_keys, checkpoint_keys)
 
+        tp_plan_warnings = _verify_tp_plan(
+            [k for k, _ in model_to_load.named_parameters()], getattr(model_to_load, "_tp_plan", None)
+        )
+        for warning in tp_plan_warnings:
+            logger.warning(warning)
```

Comment:
can't we do something like logger.warning("".join(tp_plan_warnings)) ? also we should do this in the verify tp plan IMO!